### PR TITLE
Fix ca certs

### PR DESCRIPTION
**Pull request by williamcroberts**

### CODE REVIEW

**Comment by williamcroberts** [comment]
@dkobras I am not sure why the nss-tests are failing. I moved the ca setup to a function, and set it up, but I broke something and I am not sure why, can you spot the issue?

```bash
writing new private key to 'smimeclient.key'
-----
+ echo 'Signing S/MIME certificate with test CA'
Signing S/MIME certificate with test CA
+ openssl ca -batch -keyfile ca.key -cert ca.pem -in smimeclient.csr -key '' -out smimeclient.crt -extensions smime -extfile /home/wcrobert/workspace/tpm2-pkcs11/test/integration/fixtures/xpextensions -config /home/wcrobert/workspace/tpm2-pkcs11/test/integration/fixtures/client.cnf
Using configuration from /home/wcrobert/workspace/tpm2-pkcs11/test/integration/fixtures/client.cnf
Check that the request matches the signature
Signature ok
ERROR: adding extensions in section smime
++ onerror
++ echo 'openssl ca -batch -keyfile "$CA_KEY" -cert "$CA_PEM" -in smimeclient.csr -key "$PASSWORD_CA" -out smimeclient.crt -extensions smime -extfile "$EXT_FILE" -config "$CLIENT_CNF" on line 60 failed: 1'
openssl ca -batch -keyfile "$CA_KEY" -cert "$CA_PEM" -in smimeclient.csr -key "$PASSWORD_CA" -out smimeclient.crt -extensions smime -extfile "$EXT_FILE" -config "$CLIENT_CNF" on line 60 failed: 1
```

File: test/integration/nss-tests.sh [hunk 1]
```diff
@@ -3,12 +3,10 @@
 
 set -xo pipefail
 
-CA_PEM="$TEST_FIXTURES/ca.pem"
-CA_KEY="$TEST_FIXTURES/ca.key"
-
-CLIENT_CNF="$TEST_FIXTURES/client.cnf"
-PASSWORD_CA=whatever
 EXT_FILE="$TEST_FIXTURES/smimeextensions"
+CLIENT_CNF="$TEST_FIXTURES/client.cnf"
```
Comment (dkobras) [inline test/integration/nss-tests.sh]: `EXT_FILE` and `CLIENT_CNF` are overwritten in the call to `setup_ca` below. `CLIENT_CNF` remains the same, but `EXT_FILE` points to `xpextensions` instead. That's why `openssl ca` fails later on. I'd suggest to just remove both declarations here, and replace the deference of `$EXT_FILE` in the `openssl ca` call with `$TEST_FIXTURES/smimeextensions`.

**Comment by dkobras** [comment]
@williamcroberts Yupp, `-extfile` now points to the wrong file. See my comment on the commit for a proposed fix.

File: test/integration/nss-tests.sh (hunk 1 above)
Comment (williamcroberts) [reply test/integration/nss-tests.sh]: Thank you! I missed that

**Pull request merged**